solr/solrj/src/test/org/apache/solr/common/util/EnvUtilsTest.java

```diff
@@ -111,10 +92,12 @@ public void testNotMapped() {
   @Test
   public void testOverwrite() {
     EnvUtils.setProperty("solr.overwrite", "original");
-    EnvUtils.setEnv("SOLR_OVERWRITE", "overwritten");
-    EnvUtils.init(false);
+    var env2 = Map.of("SOLR_OVERWRITE", "overwritten");
+    EnvUtils.init(false, env2);
     assertEquals("original", EnvUtils.getProperty("solr.overwrite"));
-    EnvUtils.init(true);
+    EnvUtils.init(true, env2);
     assertEquals("overwritten", EnvUtils.getProperty("solr.overwrite"));
+
+    EnvUtils.init(true, ENV); // back to how it was...
```

Comment:
Come to think of it, this is not needed since the only mutable state is sys props, which are reset not only at suite level but test level automatically by test infra